File: mycode/code/hash_leetcode.c
/***********************************************************
File name : hash_leetcode.c
Author name : xiongyansong
Email address : sanxisui@gmail.com
Create time : 2023-03-21 14:39:21
Description : 尝试学习hash查找 /看过c语言版本的了，现在用go语言写一个！
              变味了，在这里尝试c宏定义函数来了
***********************************************************/
#include <stdio.h>
// #define find_array_member(target, array)         \
//     do {                                         \
//         for (int i = 0; i < sizeof array; i++) { \
//             if (array[i] == target) {            \
//                 return i;                        \
//             }                                    \
//         }
// }
// while (0)

// typedef struct  {
//     void *array;
//     // extern len;
//     int cap
// }slice;

// func()
// {
//     return 1;
// }
int main()
{
    char a = 'c';

    printf("%ld", sizeof 77);

    int array[4] = {0, 1, 2, 3};
    int b = func();
    // int c = find_array_member(3, array);
    printf("c:%d\n", b);
}

File: mycode/new_langauge/hash_leetcode.go
/***********************************************************
File name : hash_leetcode.go
Author name : xiongyansong
Email address : sanxisui@gmail.com
Create time : 2023-03-22 15:42:14
Description : hash查找go语言版本,gg了
			  不想写了是一部分，另一部分是有那个切片好像比数组好使
			  2023-03-23 14:44:54又gg了
			  为什么数组不能a[i++]啊，真让人摸不着头脑
***********************************************************/
package main

//hash函数
func hash_func(value int) int {
	return value % 10
}

//hash查找

//插入hash表
func inster_hashelemrnt(numb int, hash_array []int) int {
	new_numb:=hash_func(numb)
	if hash_array[new_numb++] !=0 {
		hash_array[new_numb]=numb
	}else if hash_array[new_numb+1]!=0 {

	}
	return 0
}

func main() {
	var length_hash = [10]int{0}
	inster_hashelemrnt(5, length_hash[])

}
